src/promptflow/promptflow/_sdk/_orm/run_info.py

```diff
@@ -110,7 +116,16 @@ def update(
             self.end_time = end_time if isinstance(end_time, str) else end_time.isoformat()
             update_dict["end_time"] = self.end_time
         with mgmt_db_session() as session:
-            session.query(RunInfo).filter(RunInfo.name == self.name).update(update_dict)
+            # if not update system metrics, we can directly update the row;
+            # otherwise, we need to get properties first, update the dict and finally update the row
+            if system_metrics is None:
+                session.query(RunInfo).filter(RunInfo.name == self.name).update(update_dict)
+            else:
+                run_info = session.query(RunInfo).filter(RunInfo.name == self.name).first()
```

Comment:
We may lose the first changes when two transaction commits one by one, but it's a corner case, we assume it's acceptable.

Reply:
yes, I think high concurrency update on properties (system metrics only for now), should be a very edge case, so it is acceptable for me.

Reply:
added some comments on this, in case we want to completly resolve this.